File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/TypeTemplateService.java
package com.pinyougou.sellergoods.service;
import com.pinyougou.pojo.PageResult;
import com.pinyougou.pojo.TbTypeTemplate;

import java.util.List;
import java.util.Map;


/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface TypeTemplateService {

    /**
     * @Author: ZhangYu
     * @Date:2019-11-12
     * @Description: java学习
     * @Question: 查询所有的模板
     */
    List<TbTypeTemplate> findAll();

    /**
     * @Author: ZhangYu
     * @Date:2019-11-12
     * @Description: java学习
     * @Question:添加模板
     */
    void add(TbTypeTemplate template);

    /**
     * @Author: ZhangYu
     * @Date:2019-11-12
     * @Description: java学习
     * @Question:查询一个模板
     */
    TbTypeTemplate findOne(long id);

    /**
     * @Author: ZhangYu
     * @Date:2019-11-12
     * @Description: java学习
     * @Question:修改模板
     */
    void update(TbTypeTemplate template);

    /**
     * @Author: ZhangYu
     * @Date:2019-11-12
     * @Description: java学习
     * @Question:删除方法
     */
    void delete(Long[] ids);

    /**
     * @Author: ZhangYu
     * @Date:2019-11-14
     * @Description: java学习
     * @Question:通过模板id查询规格列表
     */
    List<Map> findSpecList(Long id);
}

File: pinyougou-manager-web/src/main/webapp/js/service/typeTemplateService.js
app.service("typeTemplateService",function ($http) {

    //1.查询所有的模板
    this.findAll = ()=>{
        return $http.get("/typeTemplate/findAll.do");
    };
    //2.保存添加或修改数据
    this.save = (url,entity)=>{
        return $http.post(url,entity);
    };

    //3.查找一个模板
    this.findOne = id =>{
      return $http.get("/typeTemplate/findOne.do?id="+id);
    };

    //4.删除模板
    this.dele = (ids)=>{
        return $http.get("/typeTemplate/delete.do?ids="+ids);
    }


});